soc/xtensa/esp32/esp32-mp.c

```diff
@@ -270,7 +270,7 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	}
 
 	cpus_active[0] = true;
-	cpus_active[CONFIG_MP_NUM_CPUS - 1] = true;
+	cpus_active[cpu_num - 1] = true;
```

Comment:
Shouldn't this be simply `[cpu_num]`? Since `cpu_num == 1` is the only valid number, this results in setting `cpus_active[0]` to true which is not the same behavior as before.

Reply:
We get a compile error when I set this to `[cpu_num]`, since that can be outside of the array bounds.

Reply:
Looking at it, this should be `cpus_active[cpu_num]`.  So I added ifdef protection to only build this code if CONFIG_SMP is enabled.